sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -585,34 +589,6 @@ private[hive] class HiveMetastoreCatalog(hive: HiveContext) extends Catalog with
         }
 
       case p: LogicalPlan if p.resolved => p
-
-      case p @ CreateTableAsSelect(db, tableName, child, allowExisting, None) =>
```

Comment:
Ok, I see. I'll agree with you to not mix the `CreateTableAsSelect` and the `CreateTableUsingAsSelect`. And probably we also need to refactor the `CreateTableAsSelect`, which should be used only within `HiveContext`, what do you think?